vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -168,11 +190,488 @@ public void checkAndUpdateIntegrationStatus(SDDCSoftwareConfig vc, String messag
+            if (hostInfoHypervisorVendor == null || !vendor.equals(hostInfoHypervisorVendor)) {
+               hostInfo.setHypervisorVendor(vendor);
+               needUpdate = true;
+            }
+
+            if (hardwareSummary.getNumCpuCores() != hostInfo.getCpuTotalCores()) {
+               hostInfo.setCpuTotalCores(hardwareSummary.getNumCpuCores());
+               needUpdate = true;
+            }
+
+            if (hardwareSummary.getNumCpuPkgs() != hostInfo.getCpuTotalPackages()) {
+               hostInfo.setCpuTotalPackages(hardwareSummary.getNumCpuPkgs());
+               needUpdate = true;
+            }
+
+            if (hardwareSummary.getNumCpuThreads() != hostInfo.getCpuTotalThreads()) {
+               hostInfo.setCpuTotalThreads(hardwareSummary.getNumCpuThreads());
+               needUpdate = true;
+            }
+
+            if (hardwareSummary.getCpuMhz() != hostInfo.getSingleCoreCpuMhz()) {
+               hostInfo.setSingleCoreCpuMhz(hardwareSummary.getCpuMhz());
+               needUpdate = true;
+            }
+
+            if (hardwareSummary.getMemorySize() != hostInfo.getMemoryCapacity()) {
+               hostInfo.setMemoryCapacity(hardwareSummary.getMemorySize());
+               needUpdate = true;
+            }
+         }
+      }
+
+      Map<String, HostNic> newHostNicMap = new HashMap<>();
+      Map<String, HostNic> oldHostNicMap = new HashMap<>();
+
+      NetworkInfo networkInfo = host.getConfig().getNetwork();
+      if (networkInfo != null) {
+         
+         PhysicalNic[] physicalNics = networkInfo.getPnic();
+         List<HostNic> hostNics = new ArrayList<HostNic>();
+         
+         for(PhysicalNic physicalNic : physicalNics){
+            
+            HostNic hostNic = new HostNic();
+
+            hostNic.setMacAddress(physicalNic.getMac());
+            hostNic.setDriver(physicalNic.getDriver());
+            LinkSpeedDuplex physicalNicLinkSpeed = physicalNic.getLinkSpeed();
+            if(physicalNicLinkSpeed == null) {
+               hostNic.setDuplex(false);
+               hostNic.setLinkSpeedMb(0);
+            }else {
+               hostNic.setDuplex(physicalNicLinkSpeed.isDuplex());
+               hostNic.setLinkSpeedMb(physicalNicLinkSpeed.getSpeedMb());
+            }
+            hostNic.setName(physicalNic.getDevice());
+            hostNics.add(hostNic);
+            newHostNicMap.put(physicalNic.getMac(), hostNic);
+         }
+         
+         int nicsNum = physicalNics.length;
+         if (nicsNum > 0) {
+            List<HostNic> oldNics = hostInfo.getHostNics();
+            if(oldNics == null) {
+               hostInfo.setHostNics(hostNics);
+               needUpdate = true;
+            }else {
+               for (HostNic oldhostNic : oldNics) {
+                  oldHostNicMap.put(oldhostNic.getMacAddress(), oldhostNic);
+               }
+               if (newHostNicMap.size() != oldHostNicMap.size() || !newHostNicMap.equals(oldHostNicMap)) {
```

Comment:
should check the mac address , if any change of the mac address then it will need update the nic list.